src/storage/table/column_reader_writer.cpp

```diff
@@ -434,27 +434,16 @@ void ColumnReadWriter::readFromPage(const Transaction* transaction, page_idx_t p
     fileHandleToPin->optimisticReadPage(pageIdxToPin, readFunc);
 }
 
-bool ColumnReadWriter::updatePageWithCursor(PageCursor cursor,
+void ColumnReadWriter::updatePageWithCursor(PageCursor cursor,
     const std::function<void(uint8_t*, offset_t)>& writeOp) const {
-    bool insertingNewPage = false;
     if (cursor.pageIdx == INVALID_PAGE_IDX) {
         writeOp(nullptr, cursor.elemPosInPage);
-        return false;
+        return;
     }
-
-    // The implemented mechanism for inserting new pages here doesn't work if we do concurrent
-    // writes We currently don't do concurrent writes but should also actually never hit the case
-    // where we need to insert pages either
```

Comment:
Was this comment removal intentional? Seems like it still applies to the asset?

Reply:
Yes, it was intentional; it is not meaningful any more since I removed the code that might have allowed this to add new pages to the end of the file (with the following code block removed it should now throw an exception when we attempt to read off the end of the file).
It still may be possible with the assert disabled, but it *should* never happen, is currently untested for that reason, and I don't think we will need it in future either (particularly if we want to support concurrent writes).